src/mist/io/methods.py

```diff
@@ -2598,12 +2602,16 @@ def _machine_action(user, cloud_id, machine_id, action, plan_id=None, name=None)
                         port = node_info.extra['network_settings']['Ports']['22/tcp'][0]['HostPort']
                     except KeyError:
                         port = 22
-                    machine = Machine.objects.get(cloud=cloud,
-                                                  machine_id=machine_id)
-                    for key_assoc in machine.key_associations:
-                        key_assoc.port = port
-                    machine.save()
 
+                    try:
+                        machine = Machine.objects.get(cloud=cloud,
+                                                      machine_id=machine_id)
+                    except DoesNotExist:
+                        pass
+                    else:
+                        for key_assoc in machine.key_associations:
+                            key_assoc.port = port
+                        machine.save()
```

Comment:
on the try/except a few lines before, if machine is not found, this machine.save() will break